core/controllers/story_editor.py

```diff
@@ -131,6 +131,12 @@ def put(self, story_id):
 
         self.render_json(self.values)
 
+        # Generate new TopicSummary after a Story has been edited to
+        # make sure the TopicSummaryTile displays the correct number
+        # of chapters.
+        topic_id = story.corresponding_topic_id
+        topic_services.generate_topic_summary(topic_id)
+
```

Comment:
Is it okay to put this code here in the `put` method? I originally put it in `story_services.py / _save_story()` but that causes a circular import since `topic_services` imports `story_services`.

Reply:
@aks681 PTAL!

Reply:
I'd prefer it to be in the same method as update_story, since you would never update a story without changing this, right?
How about a new function - `topic_services.update_story_and_topic_summary`, which calls `story_services.update_story` and updates topic summary, woudl that work?

Reply:
@aks681 This would still require `story_services.py` to import `topic_services.py`, right? In that case, would we have a circular import? Unless inline import statements are allowed.

Reply:
@EricZLou Sorry, I might be misunderstanding, but why would it require that? I think story_editor.py can call topic_services.update_story_and_topic_summary() which then delegates part of its work to story_services.update_story(), right?

Reply:
Though @aks681 would it be possible to prevent update_story() from being accessed/called directly by callers? Otherwise, devs might use it and bypass the function in topic_services.

Reply:
Ah I understand now -- pushing changes now.